1.8/usage/tutorials/tutorial/resources.md

```diff
@@ -0,0 +1,99 @@
+  "cpus": 1,
+  "mem": 128,
+  "disk": 0,
+  "gpus": 0,
+  ...
+```
+
+  *Note:* This specifies the allocated resources and hence the upper limit of resources which can be used by the task. This is not necessary the same as used resources, which is usually less.
+
+  ## Scaling Applications
+
+  When we need more resources for our app, we can scale in two dimensions: horizontally and vertically.
+
+  ### Scale horizontally by increasing the instance count
+
+  You can scale the instance count in two ways: scale an entire app group by a factor or directly set number of instances for an app.
+
+  * Scale dcos-101 app group. This is useful if you want to scale multiple apps in a single group:
+    * Scale up by factor 2: `dcos marathon group scale dcos-101 2`
+    * Check that both app1 and app2 have scaled: <br/>
+      `dcos marathon app list`.
+    * Scale down again: `dcos marathon group scale dcos-101 0.5`
+    * Check that both app1 and app2 have scaled: <br/>
+      `dcos marathon app list`.
+  * Set instance count for app2 directly. This is useful is you want to scale app a single app or if the factor based scaling is not applicable:
+    * `dcos marathon app update /dcos-101/app2 instances=3` Note that these are applied incrementally to an existing app definition.
+    * Check that app2 has scaled `dcos marathon app update /dcos-101/app2 instances=1`.
+    * Rescale app2 to 1 instance `dcos marathon app update /dcos-101/app2 instances=1`.
+    * Check that app2 has scaled `dcos marathon app list`.
+
+  ### Scale vertically by increasing allocated resources.
+
+  *Note*: This causes a restart of the app!
+
+  * Scale up: `dcos marathon app update /dcos-101/app2 cpus=2`.
+  *  Check that app2 has scaled `dcos marathon app list`.
+  * Scale down: `dcos marathon app update /dcos-101/app2 cpus=1`.
+
+  ## Debugging Resource Problems
+
+  * Too few resources in cluster:
+      * `dcos marathon app update /dcos-101/app2 instances=100` (potentially more if you have a large cluster).
+      * Use `dcos marathon app list` to check that the `scale` deployment is stuck.
+      * `dcos marathon deployment list`.
+      * The problem here is that there no matching resources, for example, there might be resources left for the public-slave role, but not for the default role.
+      * Solution: Add nodes or scale the app to a level at which resources are available `dcos marathon app update /dcos-101/app2 --force instances=1`. Note you have to use the `--force` flag here as the previous deployment is ongoing.
+  * Too few resources on single node. As each app is started on a single node, task resources must fit onto a single node.
+      * Update app2 to use 100 CPUs: `dcos marathon app update /dcos-101/app2 cpus=100`
+      * Use `dcos marathon app list` to check that the `restart` deployment is stuck.
+      * `dcos marathon deployment list`.
+      * The problem here is that there no resource offer being large enough to match the offer.
```

Comment:
Suggest: "The problem here is that there are no resources offers large enough to match the request."